k8-cluster/on-prem/rke2/ansible/playbook/control-plane-primary.yaml

```diff
@@ -0,0 +1,55 @@
+- hosts: control_plane_primary
+  become: yes
+  tasks:
+    - name: Generate RKE2 token
+      set_fact:
+        rke2_token: "{{ lookup('password', '/dev/null length=32 chars=hexdigits') }}"
+      run_once: true
+
+    - name: Copy RKE2 config template
+      copy:
+        src: "../../rke2-server-control-plane-primary.conf.template"
+        dest: "{{ RKE2_PATH }}/config.yaml"
+
+    - name: Replace placeholders in RKE2 config
+      lineinfile:
+        path: "{{ RKE2_PATH }}/config.yaml"
+        regexp: "^{{ item.regexp }}"
+        line: "{{ item.line }}"
+      loop:
+        - { regexp: 'token:.*', line: "token: {{ rke2_token }}" }
+        - { regexp: 'node-name:.*', line: "node-name: {{ inventory_hostname }}" }
+        - { regexp: 'node-ip:.*', line: "node-ip: {{ ansible_host }}" }
+      no_log: true 
+
+    - name: Start and enable RKE2 service
+      systemd:
+        name: rke2-server
+        enabled: yes
+        state: started
+
+    - name: Configure kubectl and kubeconfig for primary control plane node
+      shell: |
+        if [[ -f "{{ RKE2_PATH }}/rke2.yaml" ]]; then
+          cp /var/lib/rancher/rke2/bin/kubectl /bin/kubectl
+          
+          if [[ "{{ ansible_user }}" == "root" ]]; then
+            HOME_DIR="/root"
+          else
+            HOME_DIR="/home/{{ ansible_user }}"
+          fi
+
+          mkdir -p $HOME_DIR/.kube/
+          INTERNAL_IP={{ ansible_host }}
+          SUFFIX_NAME={{ inventory_hostname }}
+          CLUSTER_DOMAIN={{ cluster_domain }}
+          
+          cat "{{ RKE2_PATH }}/rke2.yaml" | sed "s/127.0.0.1/${INTERNAL_IP}/g" | sed "s/default/${CLUSTER_DOMAIN}/g" | tee -a $HOME_DIR/.kube/${CLUSTER_DOMAIN}-${SUFFIX_NAME}.yaml
+          
+          chown -R {{ ansible_user }}:{{ ansible_user }} $HOME_DIR/.kube/*
+          chmod -R 444 $HOME_DIR/.kube/*.yaml
+          chmod +x /bin/kubectl
+        fi
+
+      args:
+        executable: /bin/bash
```

Comment:
add a task to copy the kubeconfig file into the existing directory of host from where ansible command was executed.